llap-client/src/java/org/apache/hadoop/hive/llap/registry/impl/LlapRegistryService.java

```diff
@@ -132,6 +136,12 @@ private void unregisterWorker() throws IOException {
     }
   }
 
+  public void updateRegistration(Iterable<Map.Entry<String, String>> attributes) throws IOException {
+    if (isDaemon && this.registry != null) {
```

Comment:
The other register/unregister methods are private and called in serviceStart/serviceStoped only if isDaemon=true.
Since this new method is public, I wanted to make sure it has the same behaviour.